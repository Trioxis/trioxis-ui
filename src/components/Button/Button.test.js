import React from 'react';
import { Button } from './';
import { mount } from 'enzyme';

test('true is true', () => {
  const tree = mount(<Button>Wow!</Button>)
  
  expect(tree).toMatchSnapshot()
});