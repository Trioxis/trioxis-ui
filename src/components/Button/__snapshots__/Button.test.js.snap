// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render as snapshot defines 1`] = `
<defaultProps(Jss(_Button))
  Component="button"
>
  <Jss(_Button)
    Component="button"
  >
    <_Button
      Component="button"
      classes={
        Object {
          "button": "button-0-0",
        }
      }
      sheet={
        StyleSheet {
          "0.42": null,
          "attached": true,
          "classes": Object {
            "button": "button-0-0",
          },
          "deployed": true,
          "linked": false,
          "options": Object {
            "Renderer": [Function],
            "classes": Object {
              "button": "button-0-0",
            },
            "generateClassName": [Function],
            "index": -100000,
            "insertionPoint": undefined,
            "jss": Jss {
              "generateClassName": [Function],
              "options": Object {
                "Renderer": [Function],
                "createGenerateClassName": [Function],
                "plugins": Array [
                  Object {
                    "onCreateRule": [Function],
                    "onProcessRule": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                ],
              },
              "plugins": PluginsRegistry {
                "hooks": Object {
                  "onChangeValue": Array [
                    [Function],
                    [Function],
                  ],
                  "onCreateRule": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onProcessRule": Array [
                    [Function],
                    [Function],
                  ],
                  "onProcessSheet": Array [],
                  "onProcessStyle": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
              },
              "version": "8.1.0",
            },
            "meta": "Jss(_Button), Unthemed, Static",
            "parent": [Circular],
            "sheet": [Circular],
          },
          "renderer": DomRenderer {
            "element": <style
              data-jss=""
              data-meta="Jss(_Button), Unthemed, Static"
              type="text/css"
            >
              
.button-0-0 {
  color: #00868C;
  border: 3px solid #00868C;
  cursor: pointer;
  display: inline-block;
  padding: 14px 54px;
  font-size: 14px;
  background: rgba(0,0,0,0);
  font-weight: 700;
  font-family: Open Sans,sans-serif;
  line-height: 1.5;
  border-radius: 500px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-decoration: none;
}

            </style>,
            "getSelector": [Function],
            "getStyle": [Function],
            "hasInsertedRules": false,
            "setSelector": [Function],
            "setStyle": [Function],
            "sheet": [Circular],
          },
          "rules": RuleList {
            "classes": Object {
              "button": "button-0-0",
            },
            "index": Array [
              Object {
                "background": "rgba(0,0,0,0)",
                "border": "3px solid #00868C",
                "border-radius": "500px",
                "color": "#00868C",
                "cursor": "pointer",
                "display": "inline-block",
                "font-family": "Open Sans,sans-serif",
                "font-size": "14px",
                "font-weight": "700",
                "letter-spacing": "1.5px",
                "line-height": "1.5",
                "padding": "14px 54px",
                "text-decoration": "none",
                "text-transform": "uppercase",
              },
            ],
            "map": Object {
              ".button-0-0": Object {
                "background": "rgba(0,0,0,0)",
                "border": "3px solid #00868C",
                "border-radius": "500px",
                "color": "#00868C",
                "cursor": "pointer",
                "display": "inline-block",
                "font-family": "Open Sans,sans-serif",
                "font-size": "14px",
                "font-weight": "700",
                "letter-spacing": "1.5px",
                "line-height": "1.5",
                "padding": "14px 54px",
                "text-decoration": "none",
                "text-transform": "uppercase",
              },
              "button": Object {
                "background": "rgba(0,0,0,0)",
                "border": "3px solid #00868C",
                "border-radius": "500px",
                "color": "#00868C",
                "cursor": "pointer",
                "display": "inline-block",
                "font-family": "Open Sans,sans-serif",
                "font-size": "14px",
                "font-weight": "700",
                "letter-spacing": "1.5px",
                "line-height": "1.5",
                "padding": "14px 54px",
                "text-decoration": "none",
                "text-transform": "uppercase",
              },
            },
            "options": Object {
              "Renderer": [Function],
              "classes": Object {
                "button": "button-0-0",
              },
              "generateClassName": [Function],
              "index": -100000,
              "insertionPoint": undefined,
              "jss": Jss {
                "generateClassName": [Function],
                "options": Object {
                  "Renderer": [Function],
                  "createGenerateClassName": [Function],
                  "plugins": Array [
                    Object {
                      "onCreateRule": [Function],
                      "onProcessRule": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessRule": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                  ],
                },
                "plugins": PluginsRegistry {
                  "hooks": Object {
                    "onChangeValue": Array [
                      [Function],
                      [Function],
                    ],
                    "onCreateRule": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onProcessRule": Array [
                      [Function],
                      [Function],
                    ],
                    "onProcessSheet": Array [],
                    "onProcessStyle": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                  },
                },
                "version": "8.1.0",
              },
              "meta": "Jss(_Button), Unthemed, Static",
              "parent": [Circular],
              "sheet": [Circular],
            },
            "raw": Object {
              "button": Object {
                "background": "rgba(0,0,0,0)",
                "border": Array [
                  3,
                  "solid",
                  "#00868C",
                ],
                "borderRadius": 500,
                "color": "#00868C",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": Array [
                  "Open Sans",
                  "sans-serif",
                ],
                "fontSize": 14,
                "fontWeight": 700,
                "letterSpacing": 1.5,
                "lineHeight": 1.5,
                "padding": Array [
                  14,
                  54,
                ],
                "textDecoration": "none",
                "textTransform": "uppercase",
              },
            },
          },
        }
      }
    >
      <button
        className="button-0-0"
      >
        Wow!
      </button>
    </_Button>
  </Jss(_Button)>
</defaultProps(Jss(_Button))>
`;
